### ISSUE TITLE
ak.is_none doesn't work for doubly masked arrays

### ISSUE THREAD

**Issue by Dominic-Stafford** [scikit-hep/awkward#713]
Using awkward version 1.1.0rc2, I managed to create an array where some entries were None, but `ak.is_none` returned False for these entries:
```
>>> arr = ak.Array({"a": [1, 2], "b":[1, None]})
>>> arr = ak.mask(arr, [True, True])
>>> ak.is_none(arr.b)
<Array [False, False] type='2 * bool'>
```
The problem appears to come from the fact that arr.b is now doubly masked:
```
>>> arr.b.layout
<ByteMaskedArray valid_when="true">
    <mask><Index8 i="[1 1]" offset="0" length="2" at="0x000002b93290"/></mask>
    <content><IndexedOptionArray64>
        <index><Index64 i="[0 -1]" offset="0" length="2" at="0x000002b98c30"/></index>
        <content><NumpyArray format="l" shape="1" data="1" at="0x000002b96c20"/></content>
    </IndexedOptionArray64></content>
</ByteMaskedArray>
```
which doesn't seem to be possible in simpler cases, so perhaps this is the issue that needs to be solved:
```
>>> arr=ak.Array([1, None])
>>> arr = ak.mask(arr, [True, True])
>>> arr.layout
<IndexedOptionArray64>
    <index><Index64 i="[0 -1]" offset="0" length="2" at="0x000002a1f190"/></index>
    <content><NumpyArray format="l" shape="1" data="1" at="0x000002b5a8e0"/></content>
</IndexedOptionArray64>
>>> ak.is_none(arr)
<Array [False, True] type='2 * bool'>
```

Thanks

Dominic

**Comment by jpivarski** [comment]
The bug is that the double-masking occurred: there's a `simplify_optiontype` that we should be using to ensure that ByteMaskedArray of IndexedOptionArray gets collapsed down to IndexedOptionArray. I think I know where this one is—the `ByteMaskedArray::getitem_field` (`arr.b`) should probably be calling `simplify_optiontype` and it's not. I'll go look.

**Comment by jpivarski** [comment]
Should be fixed in PR #714. (I included your example in the testing suite, along with a VirtualArray version of the same, because changing the output type of `getitem_field` should be reflected in the virtual Form or you'll get a "generated the wrong Form" error.)

**Comment by Dominic-Stafford** [comment]
Thank you for fixing this so quickly!
